### PR TITLE
[Enhancement] Comprehensive support for asserting contents of lists/arrays

### LINKED ISSUE #23: [Enhancement] Comprehensive support for asserting contents of lists/arrays
**What would you like to be added**:

A way to make an assertion that the contents of a field which is an array of objects or a list:

1. has the defined portion listed somewhere/anywhere
2. has the defined portion listed in the prescribed order

**Why is this needed**:

Kubernetes has loads of resources where arrays are components. In some cases, lists are unordered (like environment variables or Policy Report results) and in others the ordering is important (like initContainers). We need a way to specify how the assertion should be performed which entails stating "this entry must be present in the collection" versus "the collection must have all these entries and they must be in this exact order".

### PR DESCRIPTION
**Pull request by shubham-cmyk**
Fixes https://github.com/kyverno/kuttl/issues/23

### CODE REVIEW

**Comment by eddycharly** [comment]
@shubham-cmyk how are we supposed to use it ?

**Comment by eddycharly** [comment]
I don't get it, can you share an example ?

**Comment by eddycharly** [comment]
ok, can we have this in the classic `assert` ?

**Comment by eddycharly** [comment]
Something like this:
```yaml
apiVersion: kuttl.dev/v1beta1
kind: TestStep
assert:
- file: policyreport.yaml
  shouldFail: false
  options:
    arrays:
    - path: /results
      strategy: anywhere
```

File: pkg/apis/testharness/v1beta1/test_types.go [hunk 1]
```diff
@@ -152,6 +159,25 @@ type TestStep struct {
 	Kubeconfig string `json:"kubeconfig,omitempty"`
 }
 
+type Assert struct {
+	// File specifies the relative or full path to the YAML containing the expected content.
+	File       string   `json:"file"`
+	ShouldFail bool     `json:"shouldFail,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/testharness/v1beta1/test_types.go:165]: i don't we need `ShouldFail` here

File: pkg/apis/testharness/v1beta1/test_types.go [hunk 2]
```diff
@@ -152,6 +159,25 @@ type TestStep struct {
...
+	Options    *Options `json:"options,omitempty"`
+}
```
Comment (eddycharly) [inline pkg/apis/testharness/v1beta1/test_types.go:167]: We need a custom `UnmarshalJSON` func

**Comment by shubham-cmyk** [comment]
Done : https://github.com/kyverno/kuttl/issues/23

**Pull request closed without merging**